### DISCUSSION TITLE
Use the message_thread_id with the bot

### DISCUSSION THREAD

**Discussion by Continueless** [windkh/node-red-contrib-telegrambot#344] [Q&A] [answered]
Hello,

how can I use message_thread_id on a sendMessage or sendPhoto? What do I have to hand over to the bot?

**Comment by windkh** [comment]
Sendtotopic.json example

**Reply by Continueless** [reply]
msg.payload.chatId = -758345456
msg.payload.type = "photo"
msg.payload.message = "Dies ist ein test."
msg.payload.content = test

this is an example for normal chat. But how can i hand out the message_thread_id to the bot ????

I have tried: msg.payload.message_thread_id and msg.payload.messagethreadid, but thats not right.

when i use the api: msg.url = "https://api.telegram.org/bot<token>/sendphoto?chat_id=-100345454543&message_thread_id=40&photo=http://bla.jpg" it works, but with the bot i cant overgive the message_thred_id

**Reply by Continueless** [reply]
>                         // 3 arguments: chatId, content, options
                        case 'createForumTopic':
                        case 'editForumTopic':
                        case 'closeForumTopic':
                        case 'reopenForumTopic':
                        case 'deleteForumTopic':
                        case 'unpinAllForumTopicMessages':
                        case 'editGeneralForumTopic':
                            // The message_thread_id must be passed in msg.payload.content: note that this is is a number not the username.
                            // Right now there is no way for resolving the user_id by username in the official API.
                            if (this.hasContent(msg)) {
                                telegramBot[type](chatId, msg.payload.content, msg.payload.options || {})
                                    .catch(function (ex) {
                                        node.processError(ex, msg, nodeSend, nodeDone);
                                    })
                                    .then(function (result) {
                                        node.processResult(result, msg, nodeSend, nodeDone);
                                    });
                            }
                            break;


this is an excerpt from the code.
It says here that it has to be passed in the payload.content. However, this is not possible for my use case because I have to transmit images and texts in the payload.content. Is it even possible to transmit the message_thread_id individually with the bot? If not, then the bot is not super group forum capable.

**Comment by Continueless** [comment, marked as answer]
msg.payload.options = {
    message_thread_id: 40
}

This ist the solution !!!

Thank you !
